utils/misc/dynamic_config.go

```diff
@@ -0,0 +1,149 @@
+package misc
```

Comment:
Is there a reason we are adding this in the `misc` package and not inside a subpackage of backendconfig, e.g. `backendconfig/dynamicconfig`?
Seems to me that it belongs there, in its own package